src/content/post/drush-make-gitignore.md

```diff
@@ -0,0 +1,52 @@
++++
+date        = "2016-04-08T01:00:00-04:00"
+title       = "Drush Make overwrites your custom .gitignore file"
+description = "An investigation about an undesired behavior: bug or feature?"
+tags        = [ "development", "drupal", "drush"  ]
+topics      = [ "Drupal" ]
+slug        = "drush-make-gitignore"
+author      = "Marcello Testi"
++++
+
+# Managing a customized .gitignore file across drush make builds
+
+## The problem
+
+**[Drush](https://github.com/drush-ops/drush) 8** overwrites the .gitignore file after a successful makefile execution.
+
+This affects complex projects where multiple dependency sources are present, and where other reasons might require a significant override of the **.gitignore** file provided by Drupal, which ends up diffed, causing `git status` to produce an ugly output, until the committed .gitignore is restored.
+
+## Origin of the problem
+
+In recent upgrades, the `--overwrite` option has been added to Drush in order to force the overwrite of existing folders (the command docs don't  mention files, althought some file reference is present in the code comments), while the default behavior is to *merge*, which means keeping existing folder content. This option has only an impact on **directories** inside the project's root and doesn't affect the files in the root.
+
+## How the problem was investigated
+
+ * I created a couple of docker containers based on SparkFabrik image ready to be used for PHP/Drupal development: https://hub.docker.com/r/sparkfabrik/docker-php-base-image/
+ * In both containers, I replaced the default Drush version (8.0.5 installed with phar) so that I could debug and alter the code on the fly
+ * In a container I installed version 6.7.0 (downloading the [release tarball](https://github.com/drush-ops/drush/releases/tag/6.7.0)) in the other I installed the "source" version of Drush 8.0.5 ([via composer](http://docs.drush.org/en/master/install-alternative/))
+ * I created a small repository containing a .gitignore slightly different than Drupal's, and a makefile with some dependencies
```

Comment:
`I created a small repository containing a .gitignore slightly different than Drupal's one, and a makefile with some dependencies`